core/kazoo_ast/src/kz_edoc_layout.erl

```diff
@@ -0,0 +1,1240 @@
+is_block_element('table') -> true;
+is_block_element('tfoot') -> true;
+is_block_element('ul') -> true;
+is_block_element('video') -> true;
+is_block_element(_) -> false.
+
+-spec par_flush([#xmlElement{}], [exporty_thing()], [exporty_thing()]) -> [exporty_thing()].
+par_flush(Es, [], Bs) ->
+    make_paraghraph(Es, [], Bs);
+par_flush(Es, As, Bs) ->
+    make_paraghraph(Es, [], Bs ++ [{p, As}]).
+
+-spec get_elem(atom(), [#xmlElement{}]) -> [#xmlElement{}].
+get_elem(Name, [#xmlElement{name = Name} = E | Es]) ->
+    [E | get_elem(Name, Es)];
+get_elem(Name, [_ | Es]) ->
+    get_elem(Name, Es);
+get_elem(_, []) ->
+    [].
+
+-spec get_attr(atom(), [#xmlAttribute{}]) -> [#xmlAttribute{}].
+get_attr(Name, [#xmlAttribute{name = Name} = A | As]) ->
+    [A | get_attr(Name, As)];
+get_attr(Name, [_ | As]) ->
+    get_attr(Name, As);
+get_attr(_, []) ->
+    [].
+
+-spec get_attrval(atom(), #xmlElement{}) -> iolist() | atom() | integer() | string().
+get_attrval(Name, #xmlElement{attributes = As}) ->
+    case get_attr(Name, As) of
+        [#xmlAttribute{value = V}] ->
+            V;
+        [] -> ""
+    end.
+
+-spec get_content(atom(), [#xmlElement{}]) -> [#xmlElement{} | #xmlText{}].
+get_content(Name, Es) ->
+    case get_elem(Name, Es) of
+        [#xmlElement{content = Es1}] ->
+            Es1;
+        [] -> []
+    end.
+
+-spec get_text(atom(), [#xmlElement{}]) -> string().
+get_text(Name, Es) ->
+    case get_content(Name, Es) of
+        [#xmlText{value = Text}] ->
+            Text;
+        [] -> ""
+    end.
+
+%%%=============================================================================
+%%% Utility functions
+%%%=============================================================================
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% @end
+%%------------------------------------------------------------------------------
+-spec is_only_whitespace(string()) -> boolean().
+is_only_whitespace([]) ->
+    true;
+is_only_whitespace([$\s | C]) ->
+    is_only_whitespace(C);
+is_only_whitespace([$\t | C]) ->
+    is_only_whitespace(C);
+is_only_whitespace([$\r | C]) ->
+    is_only_whitespace(C);
+is_only_whitespace([$\n | C]) ->
+    is_only_whitespace(C);
+is_only_whitespace(_) ->
+    false.
+
+-spec seq(fun((#xmlElement{}) -> exporty_thing()), [#xmlElement{}]) -> [exporty_thing()].
+seq(F, Es) ->
+    seq(F, Es, []).
+
+-spec seq(fun((#xmlElement{}) -> exporty_thing()), [#xmlElement{}], [string()]) -> [exporty_thing()].
+seq(F, Es, Tail) ->
+    seq(F, Es, ", ", Tail).
+
+-spec seq(fun((#xmlElement{}) -> exporty_thing()), [#xmlElement{}], string(), [string()]) -> [exporty_thing()].
+seq(F, [E], _Sep, Tail) ->
+    F(E) ++ Tail;
+seq(F, [E | Es], Sep, Tail) ->
+    F(E) ++ [Sep] ++ seq(F, Es, Sep, Tail);
+seq(_F, [], _Sep, Tail) ->
+    Tail.
+
+-spec filter_empty([{atom(), list() | iolist() | binary()}]) -> [{atom(), list() | iolist() | binary() | tuple() | atom() | boolean()}].
```

Comment:
2 questions:

1. Why not use `kz_term:proplist()` and `kz_term:proplist_property()` here and in `is_not_empty/1`?
2. Why not use `props:filter_empty/1` (or `props:filter/2` passing in `is_not_empty/1`) instead?

I think the discrepancy in the return type vs the input type is causing dialyzer (via `scripts/dialyze-changed.bash`) to complain that `kz_edoc_layout.erl:353: Function function/3 has no local return`. If you remove the call to `filter_empty/1` in that function then dialyzer is happy. When I changed the types to use our proplist types, dialyzer is happy.